File: IO/src/main/java/com/github/xzzpig/pigutils/io/GroupOutPutStream.java
package com.github.xzzpig.pigutils.io;

import java.io.IOException;
import java.io.OutputStream;
import java.util.Collections;
import java.util.LinkedList;
import java.util.List;

public class GroupOutPutStream extends OutputStream {

	List<OutputStream> outs;

	public GroupOutPutStream() {
		super();
		outs = new LinkedList<>();
	}

	public GroupOutPutStream add(OutputStream... outs) {
        Collections.addAll(this.outs, outs);
        return this;
	}

	@Override
	public void close() throws IOException {
		for (OutputStream out : outs) {
			out.close();
		}
	}

	@Override
	public void flush() throws IOException {
		for (OutputStream out : outs) {
			out.flush();
		}
	}

	protected List<OutputStream> getOutPutStreams() {
		return outs;
	}

	@Override
	public void write(int b) throws IOException {
		for (OutputStream out : outs) {
			out.write(b);
		}
	}

}

File: IO/src/main/kotlin/com/xzzpig/pigutils/io/GroupOutputStream.kt
package com.xzzpig.pigutils.io

import java.io.IOException
import java.io.OutputStream
import java.util.*

class GroupOutputStream : OutputStream() {

    private val outputStreams: MutableList<OutputStream> by lazy { LinkedList<OutputStream>() }

    fun add(vararg outs: OutputStream): GroupOutputStream {
        outputStreams.addAll(outs)
        return this
    }

    @Throws(IOException::class)
    override fun close() {
        for (out in outputStreams) {
            out.close()
        }
    }

    @Throws(IOException::class)
    override fun flush() {
        for (out in outputStreams) {
            out.flush()
        }
    }

    @Throws(IOException::class)
    override fun write(b: Int) {
        for (out in outputStreams) {
            out.write(b)
        }
    }

}
